### DISCUSSION TITLE
Import I3D Files

### DISCUSSION THREAD

**Discussion by StjerneIdioten** [StjerneIdioten/I3D-Blender-Addon#107] [Ideas & Feature Requests]
**Is your feature request related to a problem? Please describe.**
It would be nice if the plugin had the possibility of importing I3D-files to fx. rework old projects without having to start from scratch etc.

**Comment by kbrandwijk** [comment]
This would only be possible if you include decoding shapes files into this addon, which is a legal slippery slope (as it's a binary format, it would be considered reverse-engineering). The existing 'export to obj' in GE is not ideal, but I don't see a way around that. For prefabs (in #28) you will have the same problem, as almost all of them are saved as binary as well.

**Comment by StjerneIdioten** [comment]
I am fully aware of it being, as you said, a legal slippery slope, but it also very much depends on where you are located in the world and so on. Some things are for example allowed for the sake of interoperability in some places.

None the less we are talking a feature that wont get looked at until way out in the future. So maybe Giants will come around on the topic, seeing as their dlc "encryption" isn't completely tied to the binary format. Or rather it is possible to have a non encrypted binary file. We've already mentioned to them that an argument to the GE exe would be nice for making an I3D file binary in the first place, if they really wanna keep their "oh so secret" format to themselves :-P

**Comment by kbrandwijk** [comment]
I came back here, because I have started this as a separate addon. I do plan to be 100% compatible with all the additional properties from  this addon, making sure they get set on import.

**Comment by StjerneIdioten** [comment]
@kbrandwijk Hmm, the format isn't "set in stone" yet for the exporter. So you might have to change things a few times :-P Any reason for branching it out as a separate addon if you want to retain compatibility anyway?
